src/acl_kernel_if.cpp

```diff
@@ -1470,8 +1472,10 @@ void acl_kernel_if_update_status(acl_kernel_if *kern) {
 
       // Just clear the "done" bit.  The "go" bit should already have been
       // cleared, but this is harmless anyway.
-      // Since csr version 19, done bit is cleared when finish counter is read
-      if (kern->csr_version == CSR_VERSION_ID_181) {
+      // Since csr version 19, done bit is cleared when finish counter is read.
+      // Since csr version 2022.3, done bit needs to be cleared explicitly.
+      if (kern->csr_version == CSR_VERSION_ID_181 ||
+          kern->csr_version == CSR_VERSION_ID) {
```

Comment:
The above will break when the CSR version is bumped once more.
```suggestion
      // Since csr version 19, done bit is cleared when finish counter is read.
      // Since csr version 2022.3, done bit needs to be cleared explicitly.
      if (kern->csr_version == CSR_VERSION_ID_181 ||
          kern->csr_version >= CSR_VERSION_ID_2022_3) {
```